rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,151 @@
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaClientSettings(new_settings);
+            }
+            SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+                saving_to_disk_frequency = Duration::from_secs(new_settings.save_interval);
+
+                if old_settings_exit != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    log::info!("Config/settings is being saved to");
+                }
+
+                old_settings = SettingsOnDisk::RitaExitSettingsStruct(new_settings);
+            }
+        }
+
+        // debt keeper
+        if !router_storage_small {
+            let runner = AsyncSystem::new();
+            runner.block_on(async move {
+                if let Err(e) = send_debt_update() {
+                    warn!("Debt keeper update failed! {:?}", e);
+                }
+                info!("Config/settings is being saved to");
+            });
+        }
+
+        //usage tracker is invoked via trafficwatch/watch() and the block runner
+        let babel_port = settings::get_rita_common().network.babel_port;
+        trace!("Common tick!");
+
+        let res = tm_get_neighbors();
+        trace!("Currently open tunnels: {:?}", res);
+        let neighbors = res;
+        let neigh = Instant::now();
+        info!(
+            "GetNeighbors completed in {}s {}ms",
+            start.elapsed().as_secs(),
+            start.elapsed().subsec_millis()
+        );
+
+        if let Ok(mut stream) = open_babel_stream(babel_port, FAST_LOOP_TIMEOUT) {
+            if let Ok(babel_routes) = parse_routes(&mut stream) {
+                if let Err(e) = watch(babel_routes, &neighbors) {
+                    error!("Error for Rita common traffic watcher {}", e);
+                }
+                info!(
+                    "TrafficWatcher completed in {}s {}ms",
+                    neigh.elapsed().as_secs(),
+                    neigh.elapsed().subsec_millis()
+                );
+            }
+        }
+
+        let runner = AsyncSystem::new();
+        runner.block_on(async move {
+            // Check on payments, only really needs to be run this quickly
+            // on large nodes where very high variation in throughput can result
+            // in blowing through the entire grace in less than a minute
+            validate().await;
+        });
+    });
+}
+/// If the router storage space is 16MB/ it's small,
+/// we want to prevent high amounts of writes to the
+/// disk as to not run out of space.
```

Comment:
this isn't about running out of space, it's about running out of write endurance and having the drive fail.